application/src/commonMain/kotlin/com/fraktalio/fmodel/application/StateStoredAggregate.kt

```diff
@@ -97,8 +97,10 @@ interface StateOrchestratingComputation<C, S, E> : ISaga<E, C>, StateComputation
     override suspend fun S?.computeNewState(command: C): S {
         val currentState = this ?: initialState
         val events = decide(command, currentState)
-        val newState = events.fold(currentState) { s, e -> evolve(s, e) }
-        events.flatMapConcat { react(it) }.onEach { newState.computeNewState(it) }.collect()
+        var newState = events.fold(currentState) { s, e -> evolve(s, e) }
+        events.flatMapConcat { react(it) }.collect {
```

Comment:
Good catch! Thanks, Domenic.

Reply:
Yeah, that was one part of the problem we discussed last December here 
https://github.com/fraktalio/fmodel-java/issues/3#issuecomment-2528107384 

Thanks @DomenicDev

Reply:
Ahh, yes. Java port of `fmodel` might have the same problem.  Good point Rod.